include/souper/Infer/AliveDriver.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2014 The Souper Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOUPER_UTIL_ALIVE_DRIVER_H
+#define SOUPER_UTIL_ALIVE_DRIVER_H
+
+#include <unordered_map>
+#include <optional>
+#include <souper/Inst/Inst.h>
+#include "alive2/ir/function.h"
+#include "alive2/smt/smt.h"
+
+namespace souper {
+
+// TODO: Rename to AliveBuilder if we implement the ExprBuilder API in future
+class AliveDriver {
+  typedef std::unordered_map<Inst *, IR::Value *> Cache;
+public:
+  AliveDriver(Inst *LHS_, Inst *PreCondition_);
+
+  std::optional<int64_t> synthesizeConstant(souper::Inst *RHS,
+                                              std::string ConstName);
```

Comment:
indentation is funny here, I suspect you have some tabs in your patch, please be certain they are all converted to spaces